pkg/controller/controller.go

```diff
@@ -282,22 +285,30 @@ func (ctrl *resizeController) syncPVC(key string) error {
 		return err
 	}
 
-	pvc, err := ctrl.pvcLister.PersistentVolumeClaims(namespace).Get(name)
+	pvcObject, exists, err := ctrl.claims.GetByKey(key)
 	if err != nil {
-		if k8serrors.IsNotFound(err) {
-			klog.V(3).Infof("PVC %s/%s is deleted, no need to process it", namespace, name)
-			return nil
-		}
+
 		klog.Errorf("Get PVC %s/%s failed: %v", namespace, name, err)
 		return err
 	}
 
+	if !exists {
+		klog.V(3).Infof("PVC %s/%s is deleted or does not exist", namespace, name)
+		return nil
+	}
+
+	pvc, ok := pvcObject.(*v1.PersistentVolumeClaim)
+	if !ok {
+		klog.Errorf("expected PVC got: %v", pvcObject)
```

Comment:
Can we let the caller log the error? Ditto throughout

Reply:
fixed.